PubSub/src/Subscription.php

```diff
@@ -1035,6 +1061,20 @@ private function formatDeadLetterPolicyForApi(array $subscription)
         return $subscription;
     }
 
+    /**
+     * Checks if a given exception failure is because of
+     * a EOD failure reason
+     *
+     * @param ServiceException $e
+     * @return boolean
+     */
+    private function isExceptionExactlyOnce(ServiceException $e)
```

Comment:
Do we need to broaden the scope from `BadRequestException` to the parent class here? This method is only being called when a `BadRequestException` is caught.